### PR TITLE
test: add regression test for line overflow panic with `hard_tabs` (6442)

### LINKED ISSUE #6442: Compiler panic
Here's the code it died on

[https://github.com/lbfalvy/orchid/tree/e0d246fe1fa75405305ec4390ba6815049246726](https://github.com/lbfalvy/orchid/tree/e0d246fe1fa75405305ec4390ba6815049246726)

Logfile the panic asked me to attach

[rustc-ice-2025-01-12T01_00_58-10688.txt](https://github.com/user-attachments/files/18387767/rustc-ice-2025-01-12T01_00_58-10688.txt)

### PR DESCRIPTION
**Pull request by willbuckner**
The underlying panic was fixed by #7029, which computes annotation ranges in bytes rather than visual columns. Add a snapshot test to cover the original hard tabs case from #6442, and teach `warning_tests` to honor `// rustfmt-` config comments so the test can set `hard_tabs: true`.

Closes #6442

### CODE REVIEW

**Comment by ytmimi** [comment]
This is meant to fix #6442? If so, can you update the PR description with `Fixes #6442` so that GitHub properly links this PR to the issue.

**Comment by willbuckner** [comment]
> This is meant to fix #6442? If so, can you update the PR description with `Fixes #6442` so that GitHub properly links this PR to the issue.

Yes, sorry about that, added! I was too focused on the rest of my commit message and forgot to tag the issue :)

File: src/test/mod.rs [hunk 1]
```diff
@@ -263,7 +263,11 @@ fn warning_tests() {
         config.set().error_on_unformatted(true);
 
         let snapshot_name = file.file_stem().unwrap().to_str().unwrap();
-        let (parsing_errors, _, report) = format_file(file, config.clone());
+        let mut config = config.clone();
+        for (key, val) in read_significant_comments(file) {
+            config.override_value(&key, &val);
+        }
+        let (parsing_errors, _, report) = format_file(file, config);
```
Comment (ytmimi) [inline src/test/mod.rs]: I don't think this is necessary. The call to `read_config` should correctly load the significant comments.

**Review by ytmimi — CHANGES_REQUESTED** [review]
I don't think we need to update `src/test/mod.rs`

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rustfmt/7009/320de2e6d44f3190ea7cc73772e67a2ae86f5e71..598f72e55f82a6448a0efbbae75950127b566b77)*

File: src/test/mod.rs (hunk 1 above)
Comment (willbuckner) [reply src/test/mod.rs]: Fixed! You're right; I retested without it and it still works.

**Review by ytmimi — APPROVED** [review]

**Comment by ytmimi** [comment]
Thanks for originally proposing a fix, and then updating this to add the regression test!
